decentraland/unity-renderer
# fix: workaround to solve issues caused by external services putting a wearable into emotes list

## What does this PR change?

This PR introduces a Renderer workaround to the case when Kernel (or Catalyst) puts a wearable into the list of emotes before pushing this list to the Renderer. Since wearables do not have animation clips in their rendereable assets, null reference exception happened in the animation clip retrieval code. These NREs were filtered by null checks as a workaround to the source issue to at least not making a user stuck. Besides that a filtering was added into EmotesCustomizationComponentController removing any emote not having an animation clip from the backpack emotes grid view.

After fixing initial issue we figured that the same WITH_ITEMS parameter scenario produces one more issue: if you equip an emote added by WITH_ITEMS parameter and then login without WITH_ITEMS then the game will be loading for extra 45 seconds. Additional changes to emote loading promise handling code were added in this PR to fix this issue as well.

Closes #3270

Closes #3292

## How to test the changes?

1. Switch to Goerli Network
2. Go to the zone builder and create a collection with one emote and one wearable
3. Use this link template to see your emote in the world, but before using it replace [YOUR ITEM ID] with id of your new emote 
https://play.decentraland.zone/?BUILDER_SERVER_URL=https%3A%2F%2Fbuilder-api.decentraland.zone%2Fv1&NETWORK=goerli&DEBUG_MODE=true&WITH_ITEMS=[YOUR ITEM ID]&realm=artemis&position=1%2C-1&renderer-branch=fix%2Fwearable_getting_to_emotes_list_workaround
4. Find this emote in the packback, equip it, save and play it once.
5. Use the same link template but replace [YOUR ITEM ID] with id of your wearable.
6. Open backpack, check if emotes hud looks fine. 
7. Go to wearables and equip your new wearable. Check if it is equipped without issues.
8. Check if avatar is updated normally in the world
9. Check if emote that was equipped during previous session is unequipped.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
